tracing-subscriber/src/subscribe/mod.rs

```diff
@@ -719,6 +719,16 @@ where
     /// Performs late initialization when installing this subscriber as a
     /// [collector].
     ///
+    /// ## Avoiding Memory Leaks
+    ///
+    /// Subscribers should not store their own [`Dispatch`]. Doing so will prevent
+    /// them from being dropped. Instead, you may call [`Dispatch::downgrade`]
+    /// to construct a [`WeakDispatch`] (analogous to [`std::sync::Weak`]) that is
+    /// safe to stash, and can be [upgraded][`WeakDispatch::upgrade`] into a `Dispatch`.
+    ///
+    /// [`WeakDispatch`]: tracing::dispatch::WeakDispatch
+    /// [`WeakDispatch::upgrade`]: tracing::dispatch::WeakDispatch::upgrade
```

Comment:
similarly, maybe:
```suggestion
    /// Subscribers should not store the [`Dispatch`] pointing to the collector
    /// that they are a part of. Because the `Dispatch` owns the collector,
    /// storing the `Dispatch` within the collector will create a reference
    /// count cycle, preventing the `Dispatch` from ever being dropped.
    ///
    /// Instead, when it is necessary to store a cyclical reference to the
    /// `Dispatch` within a subscriber, use [`Dispatch::downgrade`] to convert a
    /// `Dispatch` into a [`WeakDispatch`]. This type is analogous to
    /// [`std::sync::Weak`], and does not create a reference count cycle. A
    /// [`WeakDispatch`] can be stored within a subscriber without causing a
    /// memory leak, and can be [upgraded] into a `Dispatch` temporarily when
    /// the `Dispatch` must be accessed by the subscriber.
    ///
    /// [`WeakDispatch`]: crate::dispatch::WeakDispatch
    /// [upgraded]: crate::dispatch::WeakDispatch::upgrade
```